pkg/river/types.go

```diff
@@ -20,7 +20,9 @@ var (
 // decode into another type or provide pre-decoding logic.
 type Unmarshaler interface {
 	// UnmarshalRiver is invoked when decoding a River value into a Go value. f
-	// should be called with a pointer to a value to decode into.
+	// should be called with a pointer to a value to decode into. UnmarshalRiver
+	// will not be called on types which are squashed into the parent struct
+	// using `river:",squash"`.
```

Comment:
Is this because the values will be squashed into the 'parent' struct already? So if you'd like to set defaults or perform validations on a squashed field, you'd have to include them in the parent's UnmarshalRiver method, right?

Reply:
> Is this because the values will be squashed into the 'parent' struct already? So if you'd like to set defaults or perform validations on a squashed field, you'd have to include them in the parent's UnmarshalRiver method, right?

This also happens in YAML, but they never explicitly document it, and I found the behavior surprising, so I wanted to document it for River. 

It's possible to support Unmarshaler on squashed structs, but it would be fairly annoying. The function passed to an Unmarshaler assumes you're unmarshaling the whole struct, and not some subset of values. For us to support Unmarshaler on struct fields, we would have to split out the attributes/blocks into two separate objects, and then call the Unmarshaler implementations twice. 

So yes, just like with YAML, unless we were to implement support for it, you'd need to set defaults and perform validations from the parent.